sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/AggregationQuerySuite.scala

```diff
@@ -545,19 +576,21 @@ abstract class AggregationQuerySuite extends QueryTest with SQLTestUtils with Te
           |  count(distinct value2),
           |  sum(distinct value2),
           |  count(distinct value1, value2),
+          |  longProductSum(distinct value1, value2),
```

Comment:
To be more specific, my question is not for the semantic of `longProductSum` since an aggregate function can define its own semantic on how to handle nulls. My question is for our rewriter. Since we use an aggregate to do distinct, for the case above, seems we will get 4 distinct input rows, which makes sense.